src/dispatching2/handler_factory.rs

```diff
@@ -0,0 +1,8 @@
+use dptree::{di::DependencyMap, Handler};
+
+/// Something that can construct a handler (for internal usage only).
+pub trait HandlerFactory {
```

Comment:
Why is this needed? 
> (for internal usage only)

And yet it's `pub`lic...

Reply:
Its implementations are generated by `#[derive(DialogueState)]`.

Reply:
Can you remove "(for internal usage only)" from documentation? It seems misleading.

Reply:
Fixed.